File: flutter/lib/models/transcript_segment.dart
/// A finalized transcription result from ASR.
class TranscriptSegment {
  final int id;
  final String text;
  final String source;
  final String speaker;
  final String lang;
  final double start;
  final double end;

  TranscriptSegment({
    required this.id,
    required this.text,
    required this.source,
    required this.speaker,
    required this.lang,
    required this.start,
    required this.end,
  });
}

File: kmp/shared/src/commonMain/kotlin/com/sondt/justtranscribe/TranscriptSegment.kt
package com.sondt.justtranscribe

import kotlinx.serialization.Serializable

/**
 * A transcribed segment returned by ASR. Mirrors the Flutter `TranscriptSegment`:
 * an incrementing [id], the recognized [text], detected [lang] (short code), and
 * the [start]/[end] stream times of the source audio.
 */
@Serializable
data class TranscriptSegment(
    val id: Int,
    val text: String,
    val lang: String,
    val start: Double,
    val end: Double,
    val source: String = "mic",
    val speaker: String = "You",
)
